jenkins/clean_project.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import collections
+import datetime
+import re
+import subprocess
+import sys
+
+
+def Instances(project, age, filt):
+  zones = collections.defaultdict(list)
+  for inst in subprocess.check_output([
+      'gcloud', 'compute', '-q',
+      'instances', 'list',
+      '--format=value(name,zone,creationTimestamp)',
+      '--filter=%s' % filt,
+      '--project=%s' % project]).split('\n'):
+    inst = inst.strip()
+    if not inst:
+      continue
+    name, zone, created_str = re.split(r'\s+', inst)
+    tfmt = 'YYYY-mm-ddTHH:MM:SS'
+    created = datetime.datetime.strptime(created_str[:len(tfmt)], '%Y-%m-%dT%H:%M:%S')
+    if created < age:
+      zones[zone].append(name)
+  return zones
+
+
+def DeleteInstances(project, zones, fake):
+  err = 0
+  for zone, instances in zones.items():
+    base = [
+        'gcloud', 'compute', '-q',
+        'instances', 'delete',
+        '--project=%s' % project,
+        '--zone=%s' % zone
+    ]
+    if fake:
```

Comment:
Hmm... I coded that up but I don't think it makes the code easier to understand or use.

I changed the default behavior to be a dry run, adding a `--delete` flag to really delete.